src/expand.rs

```diff
@@ -128,45 +129,43 @@ impl Expander {
     /// Expands the template string `template` using the syntax defined
     /// by this expander and the values of capture groups from `captures`.
     pub fn expansion(&self, template: &str, captures: &Captures<'_>) -> String {
-        let mut cursor = io::Cursor::new(Vec::with_capacity(template.len()));
+        let mut cursor = Vec::with_capacity(template.len());
         self.write_expansion(&mut cursor, template, captures)
             .expect("expansion succeeded");
-        String::from_utf8(cursor.into_inner()).expect("expansion is UTF-8")
+        String::from_utf8(cursor).expect("expansion is UTF-8")
     }
 
     /// Appends the expansion produced by `expansion` to `dst`.  Potentially more efficient
     /// than calling `expansion` directly and appending to an existing string.
     pub fn append_expansion(&self, dst: &mut String, template: &str, captures: &Captures<'_>) {
-        let pos = dst.len();
-        let mut cursor = io::Cursor::new(mem::replace(dst, String::new()).into_bytes());
-        cursor.set_position(pos as u64);
+        let mut cursor = core::mem::take(dst).into_bytes();
         self.write_expansion(&mut cursor, template, captures)
             .expect("expansion succeeded");
-        *dst = String::from_utf8(cursor.into_inner()).expect("expansion is UTF-8");
+        *dst = String::from_utf8(cursor).expect("expansion is UTF-8");
     }
 
     /// Writes the expansion produced by `expansion` to `dst`.  Potentially more efficient
     /// than calling `expansion` directly and writing the result.
     pub fn write_expansion(
         &self,
-        mut dst: impl io::Write,
+        dst: &mut Vec<u8>,
```

Comment:
This is another breaking change. @johnw42 any opinion on this?

Reply:
When I implemented this I was sure this wasn't in the public API, but I must have overlooked something. I'll gladly change this. I'm seeing multiple options here:

- Add a new function `write_expansion_vec` and gate the current one behind `cfg(feature = "std")`
- Add another `write_expansion` and activate either based on the feature
- Write a trait and implement it for either `T: std::io::Write` or `Vec<u8>` based on the feature and similarly use a type alias for the returned `Result`

I added a commit with the first option. Let me know if I should change anything.